components/layout_2020/style_ext.rs

```diff
@@ -583,7 +583,7 @@ impl ComputedValuesExt for ComputedValues {
 
     /// Whether or not the `overflow` value of this style establishes a scroll container.
     fn establishes_scroll_container(&self) -> bool {
-        self.effective_overflow().x.is_scrollable()
+        self.effective_overflow().x.is_scrollable() || self.effective_overflow().y.is_scrollable()
```

Comment:
Unnecessary: both axes need to have the same scrollability. You can add a comment (or assert) to make it clearer.